third_party/move/move-vm/runtime/src/storage/implementations/unsync_code_storage.rs

```diff
@@ -102,9 +102,13 @@ impl<M: ModuleStorage + WithRuntimeEnvironment> UnsyncCodeStorage<M> {
         let immediate_dependencies = compiled_script
             .immediate_dependencies_iter()
             .map(|(addr, name)| {
-                self.module_storage
-                    .fetch_verified_module(addr, name)
-                    .map_err(|e| expect_no_verification_errors(e))
+                if self.module_storage.check_module_exists(addr, name)? {
```

Comment:
same question as in scripts above about separate existence check - but this one is unsync so maybe the separate calls aspect doesn't matter.

Reply:
Answered above, in practice, we can transition to Result<Option<..>> APIs everywhere, and let the caller decided on what to do with the option? Might be cleaner because we sometimes want to map error to ensure there are no verification errors, but linker errors are treated differently. @gelash what do you think?

Reply:
I think that makes sense. We can do that way if nobody objects.

Reply:
I think it will be cleaner, and we save 1 executor read dispatch